src/commands/heroku.rs

```diff
@@ -412,25 +443,14 @@ pub fn deploy_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandRe
         .single::<String>()
         .expect("You must include a git ref to deploy");
 
-    let github_client = Github::new(bot_config(ctx).github_token.to_string());
+    let new_github_client = GitHubClient::new(bot_config(ctx).github_token.to_string());
 
-    if github_client.is_err() {
-        msg.reply(
-            &ctx,
-            format!(
-                "An error occured when trying to contact GitHub with your GitHub token:\n{:?}",
-                github_client.err()
-            ),
-        )?;
+    let github_request = new_github_client
+        .client
+        .get(&commit_info_url(ctx, git_ref))
+        .headers(new_github_client.headers.clone());
 
-        return Ok(());
-    }
-
-    let github_response = github_client
-        .unwrap()
-        .get()
-        .custom_endpoint(&github_ref_endpoint(bot_config(ctx), git_ref))
-        .execute::<GitHubResponse>();
+    let github_response = github_request.send();
```

Comment:
```suggestion
    let github_response = github_request.send().and_then(|res| res.error_for_status());
```

Let's error out if we receive a non-2xx status code.

Reply:
Done!